File: src/Lion.java
public class Lion extends Animal {
    private final String _lionName;

    public Lion(String lionName, String animalSpecies, short numberOfLegs, short animalAge) {
        super(animalSpecies, numberOfLegs, animalAge);
        _lionName = lionName;
    }

    @Override
    public void MakeSound() {
        System.out.printf("%s called %s makes sound <<Roar>>!%n", _animalSpecies, _lionName);
    }
}

File: src/Lion.cs
using Proiect_Animal_Sounds.src.Interfaces;

namespace Proiect_Animal_Sounds.src
{
    internal class Lion : IAnimal
    {
        public void MakeSound()
        {
            Console.WriteLine($"<{GetType().Name}> makes sound: <Roar>");
        }
    }
}
